### ISSUE TITLE
Too long email address breaks sync

### ISSUE THREAD

**Issue by ronnyjohn** [nextcloud/mail#3608]
The initial sync encountered an error and i was stuck with the loading spinner in the app. The problematic mail was a spam mail that had a really long sender address:

``OCA\Mail\Exception\ServiceException: Sync failed for 1:INBOX: An exception occurred while executing 'INSERT INTO `oc_mail_recipients` (`message_id`, `type`, `label`, `email`) VALUES(?, ?, ?, ?)' with params [145, 0, "Web.de", "celio@nk0032gy5i62diyc262borjqpr3ssgcs681x01z28lh0zx83fcf0a4ipo0wm07u717to950tn1z0v94s6ro6oh161112h70es40uxcqwbsxc6s82465v7749h125m2o538d9j1yhp125f04rfuz0b5ug78w27386vuj4tcyfb74d4mh30jaa764zzq38g7023k2lgo63zszjuj9iowij2ck3761c429t5215xtm15w3cdptxysd78ysjdpfe726bst4i8nynw8c73zt826j01rf9dedw08md2b9ibaim8tev5u6l2vhxzt53ulp623msrg8egz085b6s53yt1xafaueunsfd47bl17w18kxnjrs16j1tdek2o98ouix0h94220ai3xiwnl76q9xvc3058434d91rye57f506d7dd15td0aqv3xt1072frnjo171h2eisq92dz2ner9a65t4612wz2dp49s9l32yn13126109czer56qr95tv7322a163a1fv50c12v79bq2y22o1t2w1e20exkj04y9w56l8413g16mghrd9jokl63xrxdazk9c2d5k2k0k4fdmth6bih60cw006p0lexh08ul1b4t33j27iyigwbh791ldxm5vhxbn27s17h51itm491t5z92x6kor1g12qzef83he4031z8100u6115ooo61wu326h0yp4w1m60u79otw4xij15aw7d53zg56h0zy6fn2gu6bgxm4k10tk0n085gq5w2x631fkzd67t510ae7e219w1w4fqh8u7h2c61glgx181bmez24aqnwow00u55286z5b29kkjmslluw9m09y05f06927z5j76169kndp8b2d5v7abr5nm8xstb8y40f6ow1z9gqg32ems99q20qi9u8ny2y9ble73t8p8m2770280v42b4u8xal1f2cdkb963t5124yuho01u5ayw6n6i78w65oph3587ij6ccf2y3g1m46h0j7391n1k7s7s2em9lz65g9x3povba3x60aeva357hb87zfr9490t4i7ghu35wqo6qqhq36l452vi1zej02x5wt5s0meu4qvhw3z9w9gz7d8p30zab15910udjol0861g5fn1ogrje1051e5cni613v3w6760ath5k47zt0ku1lcww27w99cbos9h815m5ua80o9082i6xtdjk5d25pdbxi4hm590dqmpl49di64r0k9gv7e3kgffcr3jpfec20uk1jypo9dye7mahc37se0o5an4311227y0pq97wcv1y7vx41t1g099z99h1cr048snmxk4l52b809bvn0d0071w7k74xl4y44nzc4mf2l13301f1n3oi5tao1445990ap11n602a8jg5r2r261q750jrzxpx590hi8434kg3p3md61h29z6z1vt283pap55jnu54bw8onfxus1yx21cn912thc4nrwjrecozkep6hy2ek3vy5wy0cly25qd85t1715p87ls2bq06gbcym2ujxs4d06u20t0v99k73tjfgky3j2p0yh21ei32nw1xndm9712c5mq51593l8gbpk01a4qn0dx0ahw7ev4l7hb4u8onrv225nn54q5z11112i5y5022zvr7i1i0qu21u2k08816wp426q3102x1znyo1281m7bzj6v0zwi0ydw11b9mej6rj11r0uz5b0i57y68zb561st90nt250jimr0ypds82ln5b6najhires22060efk9owx3yoa3083e75elc70a9dfo2ypz8102cije5jjd0y4t0k1oq285sc39e200acb145nwi57l0v17hw5zmni5005b91xwbw73skb88id55qf0u41z4y6z110cnckb7q2tre2d9s7h06drs9t5hsk6plg856o8k02xf1o2v1gmo1v91ese3tmv18if5hco1qt4c239gr4v4f7jh4l4822uc9920kxer0k0f.goilp.club"]: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'email' at row 1``

I saw that there are other reports of bugs which broke the sync, i.e. #3421 and #2968 .

Maybe there should be a better error handling in the sync. If one email import fails it currently breaks the whole sync. All  that the user can do at this point (if he finds out what the problem is) is to use another mail client to find and delete the problematic mail. Then restart the sync and if it fails again, fix the next mail and so on...

### Mail app

**Mail app version:** 1.4.1

**Mailserver or service:** gmx.net

### Server configuration

**Operating system**: Debian 10

**Web server:** Apache

**Database:** MySQL

**PHP version:** 7.3

**Nextcloud Version:** 19.0.3

#### Client configuration
**Browser:** Vivaldi

**Operating system:** Manjaro

**Comment by ChristophWurst** [comment]
wow.


I did not expect this data. but yes, we should just truncate.

**Comment by StCyr** [comment]
truncate or ignore the message?

**Comment by ChristophWurst** [comment]
I'd say truncate, which is bad enough. Ignoring (hiding) the message from the user is even worse IMO.

**Comment by ChristophWurst** [comment]
Ref https://stackoverflow.com/questions/386294/what-is-the-maximum-length-of-a-valid-email-address for some info about *allowed* lengths.

**Comment by beccon4** [comment]
The error still occurs with the most recent version Nextcloud Hub II (23.0.0) which I run on 
- Ubuntu 20.04 in LXD container
- Ver 8.0.27-0ubuntu0.20.04.1 for Linux on x86_64 ((Ubuntu))
- PHP 8.0.14 (cli) (built: Dec 20 2021 21:22:57) ( NTS )

Unfortunately this renders the Mail app unusable with my mail account. Would it be possible to a) truncate b) drop the offending mail or at least not to break the synchronisation process? 

My user interface acts as described - the spinner circles endlessly no messages appear.

It might be of use to provide as a work around, more information on how to find the offending message among a huge stack of good ones. It is highly unlikely mail addreses with length > 244 they are of any use unless to respond to spammers :-)  It can be anywhere as the problem persists for me over a year.

`php occ mail:account:sync 1
[debug] account is up to date, skipping mailbox sync
[debug] Syncing 5
[debug] Locking mailbox 5 for new messages sync
[debug] Locking mailbox 5 for changed messages sync
[debug] Locking mailbox 5 for vanished messages sync
[debug] Running initial sync for 5
[debug] Initial sync 1:INBOX - fetch all messages from IMAP took 2s
[debug] Unlocking mailbox 5 from vanished messages sync
[debug] Unlocking mailbox 5 from changed messages sync
[debug] Unlocking mailbox 5 from new messages sync

In ImapToDbSynchronizer.php line 249:
                                                                                                                                                                         
  Sync failed for 1:INBOX: An exception occurred while executing a query: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'email' at row 1  
                                                                                                                                                                         

In ExceptionConverter.php line 114:
                                                                                                                                                
  An exception occurred while executing a query: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'email' at row 1  
                                                                                                                                                

In Exception.php line 26:
                                                                                                 
  SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'email' at row 1  
                                                                                                 

In Statement.php line 92:
                                                                                                 
  SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'email' at row 1  `


Thank you.

**Comment by miaulalala** [comment]
Yep, the issue is blocked by another issue at the moment. We're working on it :)

**Comment by brendan-pike** [comment]
This same issue effects more than just spamy email addresses, I currently can't use mail because of this:
`SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'preview_text' at row 1`

Also no idea how the preview_text is created, so not sure which email is causing it.

**Comment by ChristophWurst** [comment]
@brendan-pike please see https://github.com/nextcloud/mail/pull/6339. Your issue isn't related to the email addresses length but another attribute that needs a cut.

**Comment by ionoci** [comment]
Hi, I got the same issue I think. How may I debug/analyse wich mail it is? Or how may I help fixing this?
`[mail] Error: OCA\Mail\Exception\ServiceException: Sync failed for 1:INBOX: An exception occurred while executing a query: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'email' at row 1 at <<closure>>`

I like the app, but it's currently unuseable for me and some of my co workers.
regards
Io

**Comment by PaperTowels207** [comment]
I think this might be related to why my /var/log/maillog is receiving a crazy amount of entries, making it difficult to parse. I'm receiving the same `SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'email' at row 1` that was mentioned above in nextcloud.log.

I am not 100% certain yet, but I believe this may be what's triggering all of these entries in `maillog`, at the rate of ~2 a second.

```
Jun  6 12:43:24 hostname postfix/<port>/smtpd[pid]: connect from unknown[www.xxx.yyy.zzz]
Jun  6 12:43:24 hostname postfix/<port>/smtpd[pid]: SSL_accept error from unknown[www.xxx.yyy.zzz]: Connection reset by peer
Jun  6 12:43:24 hostname postfix/<port>/smtpd[pid]: lost connection after CONNECT from unknown[www.xxx.yyy.zzz]
Jun  6 12:43:24 hostname postfix/<port>/smtpd[pid]: disconnect from unknown[www.xxx.yyy.zzz] commands=0/0
Jun  6 12:43:25 hostname dovecot[pid2]: imap-login: Disconnected: Connection closed: read(size=1026) failed: Connection reset by peer (no auth attempts in 0 secs): user=<>, rip=www.xxx.yyy.zzz, lip=xxx
```
